syntax = "proto3";

package transport.serial;

import "map_renderer.proto";
import "transport_router.proto";
import "graph.proto";

message Stop {
    string name = 1;
    double lat = 2;
    double lng = 3;
    int32 id = 4;
}

message Distance {
    int32 from = 1;
    int32 to = 2;
    int32 val = 3;
}

message Bus {
    string name = 1;
    repeated int32 stops = 2;
    int32 last_stop = 3;
    int32 id = 4;
}

message StopBuses {
    int32 id = 1;
    repeated int32 buses = 2;
}

message Catalogue {
    repeated Stop stops = 1;
    repeated Distance distances = 2;
    repeated Bus buses = 3;
    repeated StopBuses stop_buses = 4;
}

message TransportCatalogue {
    Catalogue catalogue = 1;
    RenderSettings render_settings = 2;
    Router router = 3;
    Graph graph = 4;
}